inc/options/class-wpseo-option-wpseo.php

```diff
@@ -345,6 +347,8 @@ protected function validate_option( $dirty, $clean, $old ) {
 					break;
 
 				case 'permalink_structure':
+				case 'category_base_url':
+				case 'tag_base_url':
 					if ( isset( $dirty[ $key ] ) ) {
 						$clean[ $key ] = sanitize_option( 'permalink_structure', $dirty[ $key ] );
```

Comment:
We need to change this though, because now this case runs for category_base, tag_base, and permalink_structure but we're always updating only permalink_structure.